### PR TITLE
Extract progress logging into a separate module

### PR DESCRIPTION
**Pull request by kangasta**
Created module available in https://github.com/UpCloudLtd/progress

- [x] Rebase on top of `main`.
- [x] Update `go.mod` once there is a release available: https://github.com/UpCloudLtd/progress/pull/16

### CODE REVIEW

File: internal/commands/executor.go [hunk 1]
```diff
@@ -2,23 +2,25 @@ package commands
...
 	"github.com/UpCloudLtd/upcloud-cli/internal/output"
 	internal "github.com/UpCloudLtd/upcloud-cli/internal/service"
-	"github.com/UpCloudLtd/upcloud-cli/internal/ui"
 
 	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/service"
 	"github.com/gemalto/flume"
 )
 
 // Executor represents the execution context for commands
 type Executor interface {
```
Comment (villevsv-upcloud) [inline internal/commands/executor.go:18]: Not totally related, but I think interface is not needed as we only seem to pass around `executorImp`. Would it make sense to remove the interface and rename the struct to `Executor` at this point?

File: internal/config/progress.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"os"
+
+	"github.com/UpCloudLtd/progress"
+	"github.com/UpCloudLtd/progress/messages"
+	"github.com/jedib0t/go-pretty/v6/text"
+)
+
+func GetProgressOutputConfig() *progress.OutputConfig {
```
Comment (villevsv-upcloud) [inline internal/config/progress.go:11]: Could put these settings in `messages.GetDefaultOutputConfig()`  in `progress` repo and call directly that? Or get the current default and only override the changes here.

**Review by villevsv-upcloud — APPROVED** [review]

File: internal/commands/util.go [hunk 3]
```diff
@@ -129,10 +129,12 @@ func BoolFromString(b string) (*upcloud.Boolean, error) {
 }
 
 // HandleError logs error in livelog by setting message to msg and details to err. Returns (nil, err), where err is the err passed in as input.
-func HandleError(logline *ui.LogEntry, msg string, err error) (output.Output, error) {
-	logline.SetMessage(msg)
-	logline.SetDetails(err.Error(), "Error: ")
-	logline.MarkFailed()
+func HandleError(exec Executor, key string, err error) (output.Output, error) {
```
[the remark this answers is not in the document]
Comment (kangasta) [reply internal/commands/util.go]: Good idea, done!

File: internal/commands/executor.go (hunk 1 above)
Comment (kangasta) [reply internal/commands/executor.go:18]: I think we could create a separate ticket for that as this is already quite big PR

File: internal/config/progress.go (hunk 2 above)
Comment (kangasta) [reply internal/config/progress.go:9]: Done. This now uses default config as base and colors whole message with bright colors.

**Pull request merged**